clients/vault/src/metrics.rs

```diff
@@ -1,27 +1,522 @@
+use tokio::time::sleep;
 use tokio_metrics::TaskMetrics;
+use wallet::Balance;
 
-use runtime::prometheus::IntCounter;
-use service::Error as ServiceError;
+const SLEEP_DURATION: Duration = Duration::from_secs(5 * 60);
 
-use crate::Error;
+const CURRENCY_LABEL: &str = "currency";
+const XLM_BALANCE_TYPE_LABEL: &str = "type";
+const REQUEST_STATUS_LABEL: &str = "status";
+const TASK_NAME: &str = "task";
+const TOKIO_POLLING_INTERVAL_MS: u64 = 10000;
+const ONE: f64 = one();
 
+// Metrics are stored under the [`CURRENCY_LABEL`] key so that multiple vaults can be easily
+// monitored at the same time.
 lazy_static! {
+	pub static ref REGISTRY: Registry = Registry::new();
+	pub static ref AVERAGE_XLM_FEE: GaugeVec =
+		GaugeVec::new(Opts::new("avg_stellar_fee", "Average Stellar Fee"), &[CURRENCY_LABEL])
+			.expect("Failed to create prometheus metric");
+	pub static ref LOCKED_COLLATERAL: GaugeVec =
+		GaugeVec::new(Opts::new("locked_collateral", "Locked Collateral"), &[CURRENCY_LABEL])
+			.expect("Failed to create prometheus metric");
+	pub static ref COLLATERALIZATION: GaugeVec =
+		GaugeVec::new(Opts::new("collateralization", "Collateralization"), &[CURRENCY_LABEL])
+			.expect("Failed to create prometheus metric");
+	pub static ref REQUIRED_COLLATERAL: GaugeVec =
+		GaugeVec::new(Opts::new("required_collateral", "Required Collateral"), &[CURRENCY_LABEL])
+			.expect("Failed to create prometheus metric");
+	pub static ref MEAN_IDLE_DURATION: IntGaugeVec =
+		IntGaugeVec::new(Opts::new("mean_idle_duration_ms", "Total Idle Duration"), &[TASK_NAME])
+			.expect("Failed to create prometheus metric");
+	pub static ref MEAN_POLL_DURATION: IntGaugeVec =
+		IntGaugeVec::new(Opts::new("mean_poll_duration_ms", "Total Poll Duration"), &[TASK_NAME])
+			.expect("Failed to create prometheus metric");
+	pub static ref MEAN_SCHEDULED_DURATION: IntGaugeVec = IntGaugeVec::new(
+		Opts::new("mean_scheduled_duration_ms", "Total Scheduled Duration"),
+		&[TASK_NAME]
+	)
+	.expect("Failed to create prometheus metric");
+	pub static ref XLM_BALANCE: GaugeVec = GaugeVec::new(
+		Opts::new("stellar_balance", "Stellar Balance"),
+		&[CURRENCY_LABEL, XLM_BALANCE_TYPE_LABEL]
+	)
+	.expect("Failed to create prometheus metric");
+	pub static ref ISSUES: GaugeVec = GaugeVec::new(
+		Opts::new("issue_count", "Number of issues"),
+		&[CURRENCY_LABEL, REQUEST_STATUS_LABEL]
+	)
+	.expect("Failed to create prometheus metric");
+	pub static ref REDEEMS: GaugeVec = GaugeVec::new(
+		Opts::new("redeem_count", "Number of redeems"),
+		&[CURRENCY_LABEL, REQUEST_STATUS_LABEL]
+	)
+	.expect("Failed to create prometheus metric");
+	pub static ref NATIVE_CURRENCY_BALANCE: Gauge =
+		Gauge::new("native_currency_balance", "Native Currency Balance")
+			.expect("Failed to create prometheus metric");
+	pub static ref FEE_BUDGET_SURPLUS: GaugeVec =
+		GaugeVec::new(Opts::new("fee_budget_surplus", "Fee Budget Surplus"), &[CURRENCY_LABEL])
+			.expect("Failed to create prometheus metric");
```

Comment:
I think we can remove this one as we will not use it

Reply:
Done.